pkg/operator/ceph/cluster/nodedaemon/reconcile.go

```diff
@@ -144,7 +144,7 @@ func (r *ReconcileNode) reconcile(request reconcile.Request) (reconcile.Result,
 			logger.Errorf("more than one CephCluster found in the namespace %q, choosing the first one %q", namespace, cephCluster.GetName())
 		}
 
-		allDisabled := r.removeDisabledNodeDaemons(cephCluster.Spec, namespace)
+		allDisabled := r.removeDisabledCrashCollectorDaemons(cephCluster.Spec, namespace) && r.removeDisabledCephExporterDaemons(cephCluster.Spec, namespace)
```

Comment:
It would be good to manually test different combinations of crash collector and exporter being enabled or disabled to make sure then can be enabled or disabled independently. For example, if crash collector is never enabled, will its secret (queried on line 155 below) always exist? Looks like that's required before the node exporter would be created if enabled.

Reply:
> It would be good to manually test different combinations of crash collector and exporter being enabled or disabled to make sure then can be enabled or disabled independently. For example, if crash collector is never enabled, will its secret (queried on line 155 below) always exist? Looks like that's required before the node exporter would be created if enabled.

I did test with all scenarios and seems to be working fine.